code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,426 @@
+		if(CRASH_MARINE_MAJOR)
+			message_admins("Round finished: [MODE_CRASH_M_MAJOR]")
+			round_finished = MODE_CRASH_M_MAJOR 
+
+	return FALSE
+
+
+/datum/game_mode/crash/declare_completion()
+	. = ..()
+	to_chat(world, "<span class='round_header'>|[round_finished]|</span>")
+	to_chat(world, "<span class='round_body'>Thus ends the story of the brave men and women of the TGS Cantebury and their struggle on [SSmapping.configs[GROUND_MAP].map_name].</span>")
+	
+	// Music
+	var/xeno_track
+	var/human_track
+	switch(round_finished)
+		if(MODE_CRASH_X_MAJOR)
+			xeno_track = pick('sound/theme/winning_triumph1.ogg','sound/theme/winning_triumph2.ogg')
+			human_track = pick('sound/theme/sad_loss1.ogg','sound/theme/sad_loss2.ogg')
+		if(MODE_CRASH_M_MAJOR)
+			xeno_track = pick('sound/theme/sad_loss1.ogg','sound/theme/sad_loss2.ogg')
+			human_track = pick('sound/theme/winning_triumph1.ogg','sound/theme/winning_triumph2.ogg')
+		if(MODE_CRASH_X_MINOR)
+			xeno_track = pick('sound/theme/neutral_hopeful1.ogg','sound/theme/neutral_hopeful2.ogg')
+			human_track = pick('sound/theme/neutral_melancholy1.ogg','sound/theme/neutral_melancholy2.ogg')
+		if(MODE_CRASH_M_MINOR)
+			xeno_track = pick('sound/theme/neutral_melancholy1.ogg','sound/theme/neutral_melancholy2.ogg')
+			human_track = pick('sound/theme/neutral_hopeful1.ogg','sound/theme/neutral_hopeful2.ogg')
+		if(MODE_CRASH_DRAW_DEATH)
+			xeno_track = pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg') 
+			human_track = pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg')
+
+	for(var/i in GLOB.xeno_mob_list)
+		var/mob/M = i
+		SEND_SOUND(M, xeno_track)
+
+	for(var/i in GLOB.human_mob_list)
+		var/mob/M = i
+		SEND_SOUND(M, human_track)
+
+	for(var/i in GLOB.observer_list)
+		var/mob/M = i
+		if(ishuman(M.mind.current))
+			SEND_SOUND(M, human_track)
+			continue
+
+		if(isxeno(M.mind.current))
+			SEND_SOUND(M, xeno_track)
+			continue
+
+		SEND_SOUND(M, pick('sound/misc/gone_to_plaid.ogg', 'sound/misc/good_is_dumb.ogg', 'sound/misc/hardon.ogg', 'sound/misc/surrounded_by_assholes.ogg', 'sound/misc/outstanding_marines.ogg', 'sound/misc/asses_kicked.ogg'))
+
+	
+	log_game("[round_finished]\nGame mode: [name]\nRound time: [duration2text()]\nEnd round player population: [length(GLOB.clients)]\nTotal xenos spawned: [GLOB.round_statistics.total_xenos_created]\nTotal humans spawned: [GLOB.round_statistics.total_humans_created]")
+
+	announce_medal_awards()
+	announce_round_stats()
+
+
+
+/datum/game_mode/crash/attempt_to_join_as_larva(mob/xeno_candidate)
+	var/datum/hive_status/normal/HS = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	return HS.attempt_to_spawn_larva(xeno_candidate)
+
+
+/datum/game_mode/crash/spawn_larva(mob/xeno_candidate, mob/living/carbon/xenomorph/mother)
+	var/datum/hive_status/normal/HS = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	return HS.spawn_larva(xeno_candidate, mother)
+
+
+/datum/game_mode/crash/AttemptLateSpawn(mob/M, rank)
```

Comment:
Same, quite a bit of this could use some condensing as it's reused in the exact same manner (which makes it different from SSjob as there each bit serves a different purpose)

Reply:
this proc is already defined on the parent but the gamemode needed to adjust it so so to copy it into the gamemode.

Reply:
When you're copypasting a lot of code you can tell you're doing something wrong. You could make some more generic procs if 90% of it is the same, or call parent at a different stage.